packages/core/upload/admin/src/hooks/useRemoveFolder.js

```diff
@@ -0,0 +1,29 @@
+import { useMutation, useQueryClient } from 'react-query';
+import { useNotification } from '@strapi/helper-plugin';
+import { deleteRequest } from '../utils/deleteRequest';
+
+export const useRemoveFolder = () => {
+  const toggleNotification = useNotification();
+  const queryClient = useQueryClient();
+
+  const mutation = useMutation(id => deleteRequest('folders', id), {
+    onSuccess: () => {
+      queryClient.refetchQueries(['folders'], { active: true });
```

Comment:
Can you add rename the query to `[pluginId, 'folders']` ? So we know that there would only be one folders query in the entire project.

Reply:
Do you think we should do this for all `upload` related queries then? Meaning: `assets`, `assets-count`?